packages/babel-plugin-transform-remove-console/src/index.js

```diff
@@ -1,14 +1,67 @@
 "use strict";
 
-module.exports = function() {
+module.exports = function({ types: t }) {
   return {
     name: "transform-remove-console",
     visitor: {
       CallExpression(path) {
-        if (path.get("callee").matchesPattern("console", true)) {
-          path.remove();
+        const callee = path.get("callee");
+
+        if (!callee.isMemberExpression()) return;
+
+        if (isConsole(callee)) {
+          // console.log()
+          if (path.parentPath.isExpressionStatement()) {
+            path.remove();
+          } else {
+            path.replaceWith(createVoid0());
+          }
+        } else if (isConsoleBind(callee)) {
+          // console.log.bind()
+          path.replaceWith(createNoop());
         }
       },
+      MemberExpression: {
+        exit(path) {
+          if (isConsole(path) && !path.parentPath.isMemberExpression()) {
+            path.replaceWith(createNoop());
+          }
```

Comment:
Interesting, So, I would assume it to be replaced like this - 

```js
// in
console.foo = foo;
// out
console.foo = function () {}; 
```

which looks like the right thing to do. because it is hard to detect all usages of `console.foo` in code 

```js
const x = "foo";
const y = console;
y[x]();
```

and will lead to runtime error.